tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java

```diff
@@ -0,0 +1,450 @@
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+// TODO Add tests for copied and modified files
+@RunWith(JUnit4.class)
+public class CodeReviewServiceGetDiffFilesTest {
+  private static final String TEST_FILE = "test_file.txt";
+  private static final String TEST_FILE_CONTENTS = "Some test file contents\n";
+  private static final String FILE_IN_HEAD = "im_in_head.txt";
+  private static final String TEST_WORKSPACE = "ws1";
+  private static final String COMMIT_MESSAGE = "Some commit message";
+  private static final String REPO_ID = "startup-os";
+  private static final int DIFF_ID = 2;
+
+  private GitRepoFactory gitRepoFactory;
+  private String aaBaseFolder;
+  private String testFileCommitId;
+  private String fileInHeadCommitId;
+  private GitRepo repo;
+  private FileUtils fileUtils;
+  private CodeReviewServiceGetDiffFilesTest.TestComponent component;
+  private Server server;
+  private ManagedChannel channel;
+  private CodeReviewServiceGrpc.CodeReviewServiceBlockingStub blockingStub;
+  private CodeReviewService codeReviewService;
+  // set by createAaWorkspace
+  private String repoPath;
+
+  private FirestoreClientFactory firestoreClientFactory = mock(FirestoreClientFactory.class);
+  private FirestoreClient firestoreClient = mock(FirestoreClient.class);
+
+  @Before
+  public void setup() throws IOException {
+    Flags.parse(
+        new String[0], AuthService.class.getPackage(), CodeReviewService.class.getPackage());
+    String testFolder = Files.createTempDirectory("temp").toAbsolutePath().toString();
+    String initialRepoFolder = joinPaths(testFolder, "initial_repo");
+    aaBaseFolder = joinPaths(testFolder, "base_folder");
+
+    component =
+        DaggerCodeReviewServiceGetDiffFilesTest_TestComponent.builder()
+            .aaModule(
+                new AaModule() {
+                  @Provides
+                  @Singleton
+                  @Override
+                  @Named("Base path")
+                  public String provideBasePath(FileUtils fileUtils) {
+                    return aaBaseFolder;
+                  }
+                })
+            .build();
+    gitRepoFactory = component.getGitRepoFactory();
+    fileUtils = component.getFileUtils();
+
+    String projectId = component.getAuthService().getProjectId();
+    String token = component.getAuthService().getToken();
+    when(firestoreClientFactory.create(projectId, token)).thenReturn(firestoreClient);
```

Comment:
Instead of getting the `projectId` and `token`, can you use `any()` instead?
https://stackoverflow.com/questions/1778744/using-mockitos-generic-any-method

Reply:
Fixed.